src/characteristic.h

```diff
@@ -0,0 +1,76 @@
+/***********************************************************************
+ *
+ * Copyright (c) 2021
+ * IoTech Ltd
+ *
+ **********************************************************************/
+
+#ifndef BLE_SIM_CHARACTERISTIC_H
+#define BLE_SIM_CHARACTERISTIC_H
+
+#include <stdint.h>
+
+#include "defines.h"
+#include "descriptor.h"
+
+typedef struct characteristic_t
+{
+  char *uuid; //128-bit characteristic UUID.
+  char *service_path; //Object path of the GATT service the characteristic belongs to.
+  uint8_t * value;
+  uint32_t value_size;
+  bool notifying;
+  uint32_t flags;
+  descriptor_t *descriptors;
+  struct characteristic_t *next;
+} characteristic_t;
+
+/**
+ * Allocates memory and initialises values for a new characteristic_t 
+ * 
+ * @param uuid the uuid of the characteristic 
+ * @param service_path the path to the service the characteristic belongs to 
+ * @param descriptors characteristic descriptors 
+ * @return initialised characteristic  
+ **/
+characteristic_t *characteristic_new (const char *uuid, const char *service_path, descriptor_t *descriptors);
+
+/**
+ * Frees a characteristic_t and it's values
+ * 
+ * @param characteristic characteristic to free 
+ * @return void
+ **/
+void characteristic_free (characteristic_t *characteristic);
+
+/**
+ * Searches the characteristic for a descriptor
+ * 
+ * @param characteristic characteristic to search 
+ * @param descriptor_uuid uuid of the descriptor
+ * @return found descriptor or NULL if not found
+ **/
+descriptor_t *characteristic_get_descriptor (characteristic_t *characteristic, const char *descriptor_uuid);
+
+/**
+ * Adds a descriptor to the characteristic
+ * 
+ * @param characteristic characteristic to add the descriptor to
+ * @param descriptor descriptor to add 
+ * @return found descriptor or NULL if not found
```

Comment:
return is incorrect as it is bool and not a pointer it will either be true or false